larray/core/array.py

```diff
@@ -1603,55 +1609,110 @@ def sort_values(self, key, reverse=False):
-        >>> a
-        sex\\nat  EU  FO  BE
-              M  10   2   4
-              F   3   7   1
-        >>> a.sort_values('F')
-        sex\\nat  BE  EU  FO
-              M   4  10   2
-              F   1   3   7
-        >>> a.sort_values('F', reverse=True)
-        sex\\nat  FO  EU  BE
-              M   2  10   4
-              F   7   3   1
-        >>> b = LArray([[[10, 2, 4], [3, 7, 1]], [[5, 1, 6], [2, 8, 9]]],
-        ...            [sex, xtype, nat])
-        >>> b
-        sex  type\\nat  EU  FO  BE
-          M     type1  10   2   4
-          M     type2   3   7   1
-          F     type1   5   1   6
-          F     type2   2   8   9
-        >>> b.sort_values(('M', 'type2'))
-        sex  type\\nat  BE  EU  FO
-          M     type1   4  10   2
-          M     type2   1   3   7
-          F     type1   6   5   1
-          F     type2   9   2   8
+        sort the whole array (no key or axis given)
+        
+        >>> arr_1D = LArray([10, 2, 4], 'a=a0..a2')
+        >>> arr_1D
+        a  a0  a1  a2
+           10   2   4
+        >>> arr_1D.sort_values()
+        a  a1  a2  a0
+            2   4  10
+        >>> arr_2D = LArray([[10, 2, 4], [3, 7, 1]], 'a=a0,a1; b=b0..b2')
+        >>> arr_2D
+        a\\b  b0  b1  b2
+         a0  10   2   4
+         a1   3   7   1
+        >>> # if the array has more than one dimension, sort array with all axes combined
+        >>> arr_2D.sort_values()
+        a_b  a1_b2  a0_b1  a1_b0  a0_b2  a1_b1  a0_b0
+                 1      2      3      4      7     10
+                    
+        Sort along a given key
+                    
+        >>> # sort columns according to the values of the row associated with the label 'a1'
+        >>> arr_2D.sort_values('a1')
+        a\\b  b2  b0  b1
+         a0   4  10   2
+         a1   1   3   7
+        >>> arr_2D.sort_values('a1', reverse=True)
+        a\\b  b1  b0  b2
+         a0   2  10   4
+         a1   7   3   1
+        >>> arr_3D = LArray([[[10, 2, 4], [3, 7, 1]], [[5, 1, 6], [2, 8, 9]]],
+        ...            'a=a0,a1; b=b0,b1; c=c0..c2')
+        >>> arr_3D
+         a  b\\c  c0  c1  c2
+        a0   b0  10   2   4
+        a0   b1   3   7   1
+        a1   b0   5   1   6
+        a1   b1   2   8   9
+        >>> # sort columns according to the values of the row associated with the labels 'a0' and 'b1'
+        >>> arr_3D.sort_values(('a0', 'b1'))
+         a  b\\c  c2  c0  c1
+        a0   b0   4  10   2
+        a0   b1   1   3   7
+        a1   b0   6   5   1
+        a1   b1   9   2   8
+                   
+        Sort along an axis
+        
+        >>> arr_2D
+        a\\b  b0  b1  b2
+         a0  10   2   4
+         a1   3   7   1
+        >>> # sort values along axis 'a'
+        >>> # equivalent to sort values of each column of array arr_2D
+        >>> arr_2D.sort_values(axis='a')
+        a*\\b  b0  b1  b2
+           0   3   2   1
+           1  10   7   4
+        >>> arr_3D 
```

Comment:
I do not think having an example of sorting a 3D array along an axis in the docstring is worth its space because this is really the same thing as for the 2D array. The 3D array is fine for sorting along a key (because it is different from the 2D case -- it needs a tuple instead of a scalar). So I would move this to a unit test and **possibly** replace it by arr_2D.sort_values(axis='b')